Tests/HarmonyTests/AsyncExecutorTests.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  AsyncExecutorTests.swift
+//  HarmonyTests
+//
+//  Created by Joan Martin on 16/6/22.
+//
+
+import XCTest
+import Harmony
+import Nimble
+
+@available(iOS 13.0.0, *)
+class AsyncExecutorTests: XCTestCase {
+    
+    func test_atomic_executor_value() async throws {
+        // Given
+        let originalValue = Int.random(in: Int.min...Int.max)
+        let atomicExecutor = AsyncAtomicExecutor<Int>()
+        
+        // When
+        let value = try await atomicExecutor.submit {
+            try await Task.sleep(nanoseconds: 25*TIME_1_MILLISECOND_IN_NANOSECONDS)
+            return originalValue
+        }
+        
+        // Then
+        expect(value) == originalValue
+    }
+
+    func test_serial_execution_of_two_tasks() {
+        // Given
+        let atomicExecutor = AsyncAtomicExecutor<Int>()
+        let expectation1 = CountingTestExpectation(description: "")
+        let expectation2 = expectation(description: "")
+        
+        // When
+        Task {
```

Comment:
I would double-check that these two tasks are actually running concurrently (if that's important for the test), because if the XCTestCase class was marked as @MainActor (and it's hard to spot in Xcode sometimes) it would make this method be marked as @MainActor, and since we are using Task.init (unstructured but scoped), the actor is inherited, meaning that both tasks would be run on the main thread, defeating the purpose. 

You can always do Task.detached {...} to enforce that the actor won't be inherited.

Reply:
Could you actually refactor the code so we can see exactly what you mean?

Reply:
[Here](https://developer.apple.com/documentation/xctest/asynchronous_tests_and_expectations) it says something about:
> If your test code needs to run on the Main actor, specify @MainActor for your test method or class. If you don’t specify an actor, the test method executes on an arbitrary actor.

But probably, this is only happening if we define our Test method with `async throws` ? like we did in `test_atomic_executor_value` ?

We need to somehow verify this. Using always `Task.detached` just to be sure we're ok, doesn't look like a valid long-term option.

Reply:
If you found that documentation, then it seems OK. Traditionally the test cases would be run on the main thread and I was wondering if they imposed the same restrictions. That seems to clarify it.